src/core/core.c

```diff
@@ -192,24 +205,47 @@ void core_preinit(const char *path)
 	const char *home;
 	char *str;
 	int len;
+	int use_xdg = 0;
 
 	if (irssi_dir == NULL) {
-		home = g_get_home_dir();
-		if (home == NULL)
-			home = ".";
-
-		irssi_dir = g_strdup_printf(IRSSI_DIR_FULL, home);
+		/* check if %XDG_CONFIG_HOME/irssi exists and use to it */
+		char *dirp = g_build_filename(g_get_user_config_dir(), "irssi", NULL);
+		if (!g_file_test(dirp, G_FILE_TEST_EXISTS)) {
+			use_xdg = 1;
+			irssi_dir = g_build_filename(g_get_user_data_dir(), "irssi", NULL);
+			irssi_cache_dir = g_build_filename(g_get_user_cache_dir(), "irssi", NULL);
+			irssi_runtime_dir =
+			    g_build_filename(g_get_user_runtime_dir(), "irssi", NULL);
+		} else { /* fallback to non-xdg location */
+			home = g_get_home_dir();
+			if (home == NULL)
+				home = ".";
+			irssi_dir = g_strdup_printf(IRSSI_DIR_FULL, home);
+			/* all special XDG paths are equal to irssi_dir if XDG is not
+			 * supported */
+			irssi_cache_dir = irssi_dir;
+			irssi_runtime_dir = irssi_dir;
```

Comment:
all these memory allocations are never freed at the tear-down of this module. this should be added and maybe it's easier to strdup them instead of special-caseing the frees

Reply:
I assume that special-casing frees is meant to refer to `g_free()`
If so, `g_strdup_printf()` would still need to be freed the same way.
Can you elaborate what do you mean with strdup them?

Reply:
here these are aliases, and must only be (g_)freed once. which does happen at the end of core.c. But the same cannot be said for the variables initialised with g_build_filename, which must be g_freed individually (currently missing)

granted this is a very minor memory leak as core_deinit is called just before ending  irssi only

Reply:
Thanks for the clarification!

> granted this is a very minor memory leak as core_deinit is called just before ending irssi only

Well, it is still a memory leak. I am going to fix this.